ibis/backends/dask/execution/generic.py

```diff
@@ -0,0 +1,935 @@
+def execute_node_nullif_series_scalar(op, series, value, **kwargs):
+    return series.where(series != value)
+
+
+@execute_node.register(ops.NullIf, simple_types, dd.Series)
+def execute_node_nullif_scalar_series(op, value, series, **kwargs):
+    # TODO - not preserving the index
+    return dd.from_array(da.where(series.eq(value).values, np.nan, value))
+
+
+@toolz.curry
+def promote_to_sequence(length, obj):
+    return obj.values if isinstance(obj, dd.Series) else np.repeat(obj, length)
+
+
+def compute_row_reduction(func, value, **kwargs):
+    final_sizes = {len(x) for x in value if isinstance(x, Sized)}
+    if not final_sizes:
+        return func(value)
+    (final_size,) = final_sizes
+    raw = func(list(map(promote_to_sequence(final_size), value)), **kwargs)
+    return dd.from_array(raw).squeeze()
+
+
+# TODO - we need this, not the pandas version
+@execute_node.register(ops.Greatest, collections.abc.Sequence)
+def execute_node_greatest_list(op, value, **kwargs):
+    return compute_row_reduction(np.maximum.reduce, value, axis=0)
+
+
+# TODO - we need this, not the pandas version
+@execute_node.register(ops.Least, collections.abc.Sequence)
+def execute_node_least_list(op, value, **kwargs):
+    return compute_row_reduction(np.minimum.reduce, value, axis=0)
+
+
+# TODO - we need this, not the pandas version
+@execute_node.register(ops.Coalesce, collections.abc.Sequence)
+def execute_node_coalesce(op, values, **kwargs):
+    # TODO: this is slow
+    return compute_row_reduction(coalesce, values)
+
+
+# TODO - we need this, not the pandas version
+@execute_node.register(ops.ExpressionList, collections.abc.Sequence)
+def execute_node_expr_list(op, sequence, **kwargs):
+    # TODO: no true approx count distinct for dask, so we use exact for now
+    columns = [e.get_name() for e in op.exprs]
+    schema = ibis.schema(list(zip(columns, (e.type() for e in op.exprs))))
+    data = {col: [execute(el, **kwargs)] for col, el in zip(columns, sequence)}
+    return schema.apply_to(
+        dd.from_pandas(pd.DataFrame(data, columns=columns), npartitions=1)
+    )
+
+
+def wrap_case_result(raw: np.ndarray, expr: ir.ValueExpr):
+    """Wrap a CASE statement result in a Series and handle returning scalars.
+
+    Parameters
+    ----------
+    raw : ndarray[T]
+        The raw results of executing the ``CASE`` expression
+    expr : ValueExpr
+        The expression from the which `raw` was computed
+
+    Returns
+    -------
+    Union[scalar, Series]
+    """
+    # TODO - improve this
+    raw_1d = np.atleast_1d(raw)
+    if np.any(isnull(raw_1d)):
+        result = dd.from_array(raw_1d)
+    else:
+        result = dd.from_array(
+            raw_1d.astype(constants.IBIS_TYPE_TO_PANDAS_TYPE[expr.type()])
+        )
+    # TODO - we force computation here
+    if isinstance(expr, ir.ScalarExpr) and result.size.compute() == 1:
+        return result.head().item()
+    return result
+
+
+# TODO - we need this, not the pandas version
```

Comment:
umm, why?

Reply:
The version of `wrap_case_result` (above) is different than the pandas one

Reply:
oh ok, totally not obvious (its also NOT using a dd input, so this is odd?)

Reply:
I was mistaken here - we didn't actually need to clone these functions and it caused problems with other tests. I've removed those functions and registered the dask specific ones.